src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -253,6 +254,157 @@ public static StringValues Concat(StringValues values1, StringValues values2)
             return new StringValues(combined);
         }
 
+        public static bool Equals(StringValues left, StringValues right)
+        {
+            // .Count is an expression not a numeric so cache locally
+            var count = left.Count;
```

Comment:
Do you have any perf data that suggests this is important?

Side-note, if you didn't leave the comment, I'd just assume you were being tidy.